Xamarin.Forms.Platform.Android/Material/MaterialFrameRenderer.cs

```diff
@@ -180,129 +128,178 @@ protected virtual void OnElementChanged(ElementChangedEventArgs<Frame> e)
+				{
+					var renderer = Platform.GetRenderer(visualElement);
+					renderer?.UpdateLayout();
+				}
 			}
 		}
 
 		public override bool OnTouchEvent(MotionEvent e)
 		{
 			if (_gestureManager.OnTouchEvent(e) || base.OnTouchEvent(e))
-			{
 				return true;
-			}
 
 			return _motionEventHelper.HandleMotionEvent(Parent, e);
 		}
 
-		protected virtual void OnElementPropertyChanged(object sender, PropertyChangedEventArgs e)
+		void UpdateShadow()
 		{
-			ElementPropertyChanged?.Invoke(this, e);
+			if (_disposed || Element == null)
+				return;
 
-			if (e.PropertyName == Frame.HasShadowProperty.PropertyName)
-				UpdateShadow();
-			else if (e.PropertyName == VisualElement.BackgroundColorProperty.PropertyName)
-				UpdateBackgroundColor();
-			else if (e.PropertyName == Frame.CornerRadiusProperty.PropertyName)
-				UpdateCornerRadius();
-			else if (e.PropertyName == Frame.BorderColorProperty.PropertyName)
-				UpdateBorderColor();
+			// set the default elevation on the first time
+			if (_defaultElevation < 0)
+				_defaultElevation = CardElevation;
+
+			if (Element.HasShadow)
+				CardElevation = _defaultElevation;
+			else
+				CardElevation = 0f;
 		}
 
-		void UpdateBackgroundColor()
+		void UpdateCornerRadius()
 		{
-			if (_disposed)
+			if (_disposed || Element == null)
 				return;
 
-			Color bgColor = Element.BackgroundColor;
+			var cornerRadius = Element.CornerRadius;
+			if (cornerRadius < 0f && _defaultCornerRadius < 0f)
+				return;
 
-			int[][] States =
-			{
-				new int[0]
-			};
-
-			CardBackgroundColor = new global::Android.Content.Res.ColorStateList
-				(
-					States,
-					new int[] { bgColor.IsDefault ? AColor.White : bgColor.ToAndroid() }
-				);
-			
+			if (_defaultCornerRadius < 0f)
+				_defaultCornerRadius = Radius;
+
+			if (cornerRadius < 0f)
+				Radius = _defaultCornerRadius;
+			else
+				Radius = (int)Context.ToPixels(cornerRadius);
+
+			UpdateBorder();
 		}
 
-		void UpdateBorderColor()
+		void UpdateBorder()
 		{
-			if (_disposed)
+			if (_disposed || Element == null)
 				return;
 
-			Color borderColor = Element.BorderColor;
+			var borderColor = Element.BorderColor;
+			if (borderColor.IsDefault && _defaultStrokeColor == null && _defaultStrokeWidth < 0f)
+				return;
+
+			if (_defaultStrokeColor == null)
+				_defaultStrokeColor = StrokeColor;
+			if (_defaultStrokeWidth < 0)
+				_defaultStrokeWidth = StrokeWidth;
+
 			if (borderColor.IsDefault)
 			{
-				StrokeColor = AColor.Transparent;
-				StrokeWidth = 0;
+				StrokeColor = _defaultStrokeColor.Value;
+				StrokeWidth = _defaultStrokeWidth;
 			}
 			else
 			{
 				StrokeColor = borderColor.ToAndroid();
-				StrokeWidth = 25;
+				StrokeWidth = (int)Context.ToPixels(1);
 			}
 
+			// update the native and forms view with the border
+			SetContentPadding(StrokeWidth, StrokeWidth, StrokeWidth, StrokeWidth);
+			if (Element is IFrameController frame)
+				frame.SetContentPadding(Context.FromPixels(StrokeWidth));
```

Comment:
This chunk of code is probably going to go away with AndroidX as they have moved the contents into a separate view with it's own padding.